From 81ee0ee9a7ad84994376c2a5901046fe43a85d60 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Tue, 10 Dec 2013 14:23:47 +0000
Subject: [PATCH 067/351] ASoC: arizona: Increase MAX_DAI

MAX_DAI was not increased when the debug capture DAI drivers were added.
This patch fixes this and adds a build check to ensure this doesn't
happen in future updates.

Change-Id: Ia7bc14ac72fc76d8ea8463d8968ae82235b22185
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 sound/soc/codecs/arizona.h |    2 +-
 sound/soc/codecs/florida.c |    2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/sound/soc/codecs/arizona.h b/sound/soc/codecs/arizona.h
index e35d0eb..dadce80 100644
--- a/sound/soc/codecs/arizona.h
+++ b/sound/soc/codecs/arizona.h
@@ -57,7 +57,7 @@
 #define ARIZONA_CLK_98MHZ  5
 #define ARIZONA_CLK_147MHZ 6
 
-#define ARIZONA_MAX_DAI  8
+#define ARIZONA_MAX_DAI  10
 #define ARIZONA_MAX_ADSP 4
 
 struct arizona;
diff --git a/sound/soc/codecs/florida.c b/sound/soc/codecs/florida.c
index b340ead..2920b9c 100644
--- a/sound/soc/codecs/florida.c
+++ b/sound/soc/codecs/florida.c
@@ -2048,6 +2048,8 @@ static int florida_probe(struct platform_device *pdev)
 	struct florida_priv *florida;
 	int i, ret;
 
+	BUILD_BUG_ON(ARRAY_SIZE(florida_dai) > ARIZONA_MAX_DAI);
+
 	florida = devm_kzalloc(&pdev->dev, sizeof(struct florida_priv),
 			      GFP_KERNEL);
 	if (florida == NULL)
-- 
1.7.9.5

